// Code generated by gofhir. DO NOT EDIT.
// Source: FHIR StructureDefinitions (backbone elements)
// Package: r4

package r4

// MolecularSequenceQuality represents the MolecularSequence.quality backbone element.
// An set of value as quality of sequence
type MolecularSequenceQuality struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// indel | snp | unknown
	Type *QualityType `json:"type,omitempty"`
	// Standard sequence for comparison
	StandardSequence *CodeableConcept `json:"standardSequence,omitempty"`
	// Start position of the sequence
	Start *int `json:"start,omitempty"`
	// End position of the sequence
	End *int `json:"end,omitempty"`
	// Quality score for the comparison
	Score *Quantity `json:"score,omitempty"`
	// Method to get quality
	Method *CodeableConcept `json:"method,omitempty"`
	// True positives from the perspective of the truth data
	TruthTP *float64 `json:"truthTP,omitempty"`
	// True positives from the perspective of the query data
	QueryTP *float64 `json:"queryTP,omitempty"`
	// False negatives
	TruthFN *float64 `json:"truthFN,omitempty"`
	// False positives
	QueryFP *float64 `json:"queryFP,omitempty"`
	// False positives where the non-REF alleles in the Truth and Query Call Sets match
	GtFP *float64 `json:"gtFP,omitempty"`
	// Precision of comparison
	Precision *float64 `json:"precision,omitempty"`
	// Recall of comparison
	Recall *float64 `json:"recall,omitempty"`
	// F-score
	FScore *float64 `json:"fScore,omitempty"`
	// Receiver Operator Characteristic (ROC) Curve
	Roc *MolecularSequenceQualityRoc `json:"roc,omitempty"`
}

// MolecularSequenceQualityRoc represents the MolecularSequence.quality.roc backbone element.
// Receiver Operator Characteristic (ROC) Curve
type MolecularSequenceQualityRoc struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// Genotype quality score
	Score []int `json:"score,omitempty"`
	// Roc score true positive numbers
	NumTP []int `json:"numTP,omitempty"`
	// Roc score false positive numbers
	NumFP []int `json:"numFP,omitempty"`
	// Roc score false negative numbers
	NumFN []int `json:"numFN,omitempty"`
	// Precision of the GQ score
	Precision []float64 `json:"precision,omitempty"`
	// Sensitivity of the GQ score
	Sensitivity []float64 `json:"sensitivity,omitempty"`
	// FScore of the GQ score
	FMeasure []float64 `json:"fMeasure,omitempty"`
}

// MolecularSequenceReferenceSeq represents the MolecularSequence.referenceSeq backbone element.
// A sequence used as reference
type MolecularSequenceReferenceSeq struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// Chromosome containing genetic finding
	Chromosome *CodeableConcept `json:"chromosome,omitempty"`
	// The Genome Build used for reference, following GRCh build versions e.g. 'GRCh 37'
	GenomeBuild *string `json:"genomeBuild,omitempty"`
	// sense | antisense
	Orientation *OrientationType `json:"orientation,omitempty"`
	// Reference identifier
	ReferenceSeqId *CodeableConcept `json:"referenceSeqId,omitempty"`
	// A pointer to another MolecularSequence entity as reference sequence
	ReferenceSeqPointer *Reference `json:"referenceSeqPointer,omitempty"`
	// A string to represent reference sequence
	ReferenceSeqString *string `json:"referenceSeqString,omitempty"`
	// watson | crick
	Strand *StrandType `json:"strand,omitempty"`
	// Start position of the window on the  reference sequence
	WindowStart *int `json:"windowStart,omitempty"`
	// End position of the window on the reference sequence
	WindowEnd *int `json:"windowEnd,omitempty"`
}

// MolecularSequenceRepository represents the MolecularSequence.repository backbone element.
// External repository which contains detailed report related with observedSeq in this resource
type MolecularSequenceRepository struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// directlink | openapi | login | oauth | other
	Type *RepositoryType `json:"type,omitempty"`
	// URI of the repository
	Url *string `json:"url,omitempty"`
	// Repository's name
	Name *string `json:"name,omitempty"`
	// Id of the dataset that used to call for dataset in repository
	DatasetId *string `json:"datasetId,omitempty"`
	// Id of the variantset that used to call for variantset in repository
	VariantsetId *string `json:"variantsetId,omitempty"`
	// Id of the read
	ReadsetId *string `json:"readsetId,omitempty"`
}

// MolecularSequenceStructureVariant represents the MolecularSequence.structureVariant backbone element.
// Structural variant
type MolecularSequenceStructureVariant struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// Structural variant change type
	VariantType *CodeableConcept `json:"variantType,omitempty"`
	// Does the structural variant have base pair resolution breakpoints?
	Exact *bool `json:"exact,omitempty"`
	// Structural variant length
	Length *int `json:"length,omitempty"`
	// Structural variant outer
	Outer *MolecularSequenceStructureVariantOuter `json:"outer,omitempty"`
	// Structural variant inner
	Inner *MolecularSequenceStructureVariantInner `json:"inner,omitempty"`
}

// MolecularSequenceStructureVariantInner represents the MolecularSequence.structureVariant.inner backbone element.
// Structural variant inner
type MolecularSequenceStructureVariantInner struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// Structural variant inner start
	Start *int `json:"start,omitempty"`
	// Structural variant inner end
	End *int `json:"end,omitempty"`
}

// MolecularSequenceStructureVariantOuter represents the MolecularSequence.structureVariant.outer backbone element.
// Structural variant outer
type MolecularSequenceStructureVariantOuter struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// Structural variant outer start
	Start *int `json:"start,omitempty"`
	// Structural variant outer end
	End *int `json:"end,omitempty"`
}

// MolecularSequenceVariant represents the MolecularSequence.variant backbone element.
// Variant in sequence
type MolecularSequenceVariant struct {
	// Unique id for inter-element referencing
	Id *string `json:"id,omitempty"`
	// Additional content defined by implementations
	Extension []Extension `json:"extension,omitempty"`
	// Extensions that cannot be ignored even if unrecognized
	ModifierExtension []Extension `json:"modifierExtension,omitempty"`
	// Start position of the variant on the  reference sequence
	Start *int `json:"start,omitempty"`
	// End position of the variant on the reference sequence
	End *int `json:"end,omitempty"`
	// Allele that was observed
	ObservedAllele *string `json:"observedAllele,omitempty"`
	// Allele in the reference sequence
	ReferenceAllele *string `json:"referenceAllele,omitempty"`
	// Extended CIGAR string for aligning the sequence with reference bases
	Cigar *string `json:"cigar,omitempty"`
	// Pointer to observed variant information
	VariantPointer *Reference `json:"variantPointer,omitempty"`
}
